lib/driverProviders/sauce.ts

```diff
@@ -27,33 +25,37 @@ export class Sauce extends DriverProvider {
    * Hook to update the sauce job.
    * @public
    * @param {Object} update
-   * @return {q.promise} A promise that will resolve when the update is complete.
+   * @return {Promise} A promise that will resolve when the update is complete.
    */
-  updateJob(update: any): q.Promise<any> {
+  updateJob(update: any): Promise<any> {
     let deferredArray = this.drivers_.map((driver: WebDriver) => {
-      let deferred = q.defer();
+      let deferredResolve: () => void;
```

Comment:
Similar to below, we should just return a promise that wraps this code instead of using a deferred